topostats/processing.py

```diff
@@ -247,16 +255,120 @@ def process_scan(
                     elif grains_config["direction"] == "lower":
                         tracing_stats_df = tracing_stats["lower"].df
                     LOGGER.info(f"[{filename}] : Combining {direction} grain statistics and dnatracing statistics")
-                    # NB - Merge on molecule and threshold because we may have upper and lower molecueles which gives
-                    #      duplicate molecule numbers as they are processed separately
-                    results = grainstats_df.merge(tracing_stats_df, on=["molecule_number", "threshold"])
+                    # NB - Merge on image, molecule and threshold because we may have upper and lower molecueles which
+                    #      gives duplicate molecule numbers as they are processed separately
+                    results = grainstats_df.merge(tracing_stats_df, on=["image", "threshold", "molecule_number"])
                 else:
+                    LOGGER.info(f"[{filename}] Calculation of DNA Tracing disabled, returning grainstats data frame.")
                     results = grainstats_df
                     results["image"] = filename
```

Comment:
Might want to remove this as we found out it is redundant earlier? I think you may have already done this locally.

Reply:
D'oh, I forgot to push the removals last night, just pushed them.